python/ray/tests/test_asyncio.py

```diff
@@ -304,15 +304,22 @@ def interceptor(e):
     ray.worker._unhandled_error_handler = interceptor
     x1 = f.remote()
     del x1
-    wait_for_condition(lambda: num_exceptions == 1)
+    if client_test_enabled():
+        # In client mode, x1 may not have received the underlying exception
+        # when deleted, so no exception is raised.
+        prev_num_exceptions = num_exceptions
```

Comment:
is this a bug? Should it behave like native ray?

Reply:
I think not really. Exception handling and behavior during {Client}ObjectRef destruction is not a public API. Client uses async communication with cluster, but scripts running in cluster can communicate with workers synchronously.